integration/testutil_soci.go

```diff
@@ -91,12 +96,21 @@ func buildIndex(sh *shell.Shell, src imageInfo, opt ...indexBuildOption) string
 		o(&indexBuildConfig)
 	}
 	opts := encodeImageInfo(src)
+
+	var createArgs []string
+	if indexBuildConfig.supportLegacyRegistry {
+		createArgs = append(createArgs, "--legacy-registry")
+	}
```

Comment:
nit: move this to after line 106 (and change `=` to `:=` in line 106) to remove explicit declare in line 100 and colocate `createArgs` logic